enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,343 @@
+)
+```
+
+#### Resource provided to the installer
+
+The users provide a load balancer type to the installer.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    lbType: nlb
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+
+#### Resources created by the installer
+
+- The installer will create infrastructure cluster object with the load balancer type defined in it.
+
+```yaml
+ oc edit ingress.config/cluster -o yaml
+ # Please edit the object below. Lines beginning with a '#' will be ignored,
+ # and an empty file will abort the edit. If an error occurs while saving this file will be
+ # reopened with the relevant failures.
+ #
+ apiVersion: config.openshift.io/v1
+ kind: Ingress
+ metadata:
+   creationTimestamp: "2022-08-30T01:03:17Z"
+   generation: 11
+   name: cluster
+   resourceVersion: "60888"
+   uid: de89cc95-449e-4ae5-a837-f4171fe61e36
+ spec:
+   domain: apps.testnlbapi.devcluster.openshift.com
+   loadbalancer:
+     platform:
+       aws:
+         type: NLB
+       type: AWS
+ status:
+   componentRoutes:
+```
+
+- The cluster-ingress-operator will check the cluster ingress object for the lbType and reconcile to create an default ingress controller
+  with the specified lbType.
+
+#### Destroying cluster
+
+- None
+
+
+### Risks and Mitigations
+
+- Users need to put either nlb or elb as input or else the LB creation will fail.
+
+- To mitigate this we need to validate the input.
+
+### Drawbacks
+
+None
+
+## Design Details
+
+### Test Plan
+
+- Set the awsLBType field in the install-config.yaml either to nlb or elb and see if the default ingress controller gets created with the desired LB type.
+- Delete the default ingress controller CR and see if the default ingress controller will be created using the LB type mentioned during installation.
+- Unit tests and E2E tests will be written.
+
+### Graduation Criteria
+
+*This enhancement will follow standard graduation criteria.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- Community Testing
+- Sufficient time for feedback
+- Upgrade, Downgrade and scale testing are not relevant to this enhancement
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+
+#### Removing a deprecated feature
+None
+
+### Upgrade / Downgrade Strategy
+Upgrade won't have issues as far as the installer and OCP version(as ingress operator has the fix) have the feature.
+
+Downgrade to a version of installer and OCP will create CLB/ELB by default.
+
+### Version Skew Strategy
+Not applicable.
+
+### Operational Aspects of API Extensions
+None
+
+#### Failure Modes
+None
```

Comment:
Maybe this would be a good place to put the possible failure you mentioned in the "Risks and Mitigations" section: 'Users need to put either "NLB" or "Classic" for `lbType` or omit the field or else the install-config will fail validation.'